src/handlers/add.ts

```diff
@@ -13,9 +13,14 @@ export namespace AddHandler {
 
     const id = nanoid();
     const allTasks: Array<Task> = await getTasks();
-    const by = args.by ? new Date(args.by).toISOString() : undefined;
 
-    allTasks.push({ id: id, done: false, description: args.description, by: by });
+    try {
+      const by = args.by ? new Date(args.by).toISOString() : undefined;
+    } catch (e: unknown) {
+      console.error("Oops! Please type valid date");
+      process.exit(1);
+    }
```

Comment:
Are you sure this works? Variables defined inside a block can only be used on that block and children blocks.

Reply:
I have checked by adding and listing before pushing here, but I realized I was not checking properly. Could I fix these in a new issue #13 instead ?

Reply:
It seems like I didn't check if this works properly before pushing. Now I fixed!